kotskinds/apis/kots/v1beta1/helmchart_types_test.go

```diff
@@ -339,3 +342,235 @@ func Test_MappedChartValueGetValue(t *testing.T) {
+								},
+							},
+						},
+					},
+				},
+			},
+			overlayValues: map[string]MappedChartValue{
+				"storage": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"postgres": &MappedChartValue{
+							valueType: "children",
+							children: map[string]*MappedChartValue{
+								"enabled": &MappedChartValue{
+									boolValue: true,
+									valueType: "bool",
+								},
+								"replacementtest": &MappedChartValue{
+									strValue:  "somethingOverwritten",
+									valueType: `string`,
+								},
+							},
+						},
+					},
+				},
+			},
+			expect: map[string]MappedChartValue{
+				"storage": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"postgres": &MappedChartValue{
+							valueType: "children",
+							children: map[string]*MappedChartValue{
+								"enabled": &MappedChartValue{
+									boolValue: true,
+									valueType: "bool",
+								},
+								"replacementtest": &MappedChartValue{
+									strValue:  "somethingOverwritten",
+									valueType: `string`,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "with-deep-children-missing-overlay",
+			baseValues: map[string]MappedChartValue{
+				"storage": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"postgres": &MappedChartValue{
+							valueType: "children",
+							children: map[string]*MappedChartValue{
+								"enabled": &MappedChartValue{
+									boolValue: true,
+									valueType: "bool",
+								},
+								"replacementtest": &MappedChartValue{
+									strValue:  "somethinghello",
+									valueType: `string`,
+								},
+							},
+						},
+					},
+				},
+			},
+			overlayValues: map[string]MappedChartValue{
+				"storage": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"postgres": &MappedChartValue{
+							valueType: "children",
+							children: map[string]*MappedChartValue{
+								"enabled": &MappedChartValue{
+									boolValue: true,
+									valueType: "bool",
+								},
+							},
+						},
+					},
+				},
+			},
+			expect: map[string]MappedChartValue{
+				"storage": MappedChartValue{
+					valueType: "children",
+					children: map[string]*MappedChartValue{
+						"postgres": &MappedChartValue{
+							valueType: "children",
+							children: map[string]*MappedChartValue{
+								"enabled": &MappedChartValue{
+									boolValue: true,
+									valueType: "bool",
+								},
+								"replacementtest": &MappedChartValue{
+									strValue:  "somethinghello",
+									valueType: `string`,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			actual := MergeHelmChartValues(test.baseValues, test.overlayValues)
+			deep.CompareUnexportedFields = true
+			diff := deep.Equal(&actual, &test.expect)
```

Comment:
why not use `reflect.DeepEqual`? no need for another library

Reply:
I had seen deep.Equal used in many other projects -- I think the biggest advantage is that it prints a pretty diff

Reply:
@laverya was the last one I used it w/ -- are we still into this library?